src/chocolatey.tests/infrastructure/guards/EnsureSpecs.cs

```diff
@@ -42,19 +66,80 @@ public override void Because()
             }
 
             [Fact]
-            public void should_return_a_type_of_string_for_ensuring()
+            public void should_return_a_ensure_string_type()
             {
-                result.ShouldBeType<Ensure<string>>();
+                result.ShouldBeType<EnsureString>();
             }
 
             [Fact]
             public void should_have_the_value_specified()
             {
-                var bobEnsure = result as Ensure<string>;
+                var bobEnsure = result as EnsureString;
                 bobEnsure.Value.ShouldEqual(bob);
             }
         }
 
+        public class when_using_EnsureString : EnsureSpecsBase
+        {
+            public override void Because()
+            {
+            }
+
+            [Fact]
+            public void when_testing_a_string_against_null_value_should_fail()
+            {
+                string test = null;
+
+                Action a = () => Ensure.that(() => test).is_not_null_or_whitespace();
+
+                a.ShouldThrow<ArgumentNullException>();
+            }
+
+            [Fact]
+            public void when_testing_a_string_against_an_empty_value_should_fail()
+            {
+                Action a = () => Ensure.that(() => string.Empty).is_not_null_or_whitespace();
+
+                a.ShouldThrow<ArgumentException>();
+            }
+
+            [Fact]
+            public void when_testing_a_string_against_a_whitespace_value_should_fail()
+            {
+                var test = "      ";
+
+                Action a = () => Ensure.that(() => test).is_not_null_or_whitespace();
+
+                a.ShouldThrow<ArgumentException>();
+            }
+
+            [Fact]
+            public void when_testing_a_string_against_a_non_empty_value_should_pass()
+            {
+                var test = "some value";
+
+                Ensure.that(() => test).is_not_null_or_whitespace();
+            }
+
+            [Fact]
+            public void when_testing_a_string_without_expected_extension_should_fail()
+            {
+                var test = "some-file.png";
+
+                Action a = () => Ensure.that(() => test).has_any_extension(".jpg", ".bmp", ".gif");
+
+                a.ShouldThrow<ArgumentException>();
+            }
+
+            [Fact]
+            public void when_testing_a_string_with_expected_extension_should_pass()
+            {
+                var test = "some-file.png";
+
+                Ensure.that(() => test).has_any_extension(".jph", ".bmp", ".gif", ".png");

```

Comment:
```suggestion
                Ensure.that(() => test).has_any_extension(".jpg", ".bmp", ".gif", ".png");

```

This has no bearing on anything, just noticed that it wasn't the same as the above test.

Reply:
I did notice the type, but since it didn't matter I didn't bother fixing it.

Now that you've commented on it though, I'll make the change